### ISSUE TITLE
Admin Organize Tables Lack Accessible Design

### ISSUE THREAD

**Issue by avaaec** [pressbooks/pressbooks#1784]
There are many a11y barriers in the current Organize page. 

## Specific problems which should be addressed
* Row headers on the Organize page are not identified as headers even though they define the entire row. This means that when screen readers read out the contents of one of the checkbox cells they simply read it as a checkbox and do not associate it with the related section. Row headers should have role=”rowheader”  
![Screenshot from 2021-10-12 15-52-01](https://user-images.githubusercontent.com/13485451/137039491-4c186af5-7dde-46b2-ad1d-94f7674059a6.png)
* The links ("Edit", "Trash", "View", "Move Up", "Move Down") in the row header cells can only be viewed on hover. While these elements still receive keyboard focus, they are not visible when using a keyboard. Note that similar functionality on other pages, such as the Contributors page, functions correctly and the options are revealed when users tab through the page. 
* The buttons in the ```<tfoot>``` of each table add an additional empty row to every table, which is confusing. These cells should not have role=”columnheader”. Ideally, the Add links should exist outside of the table programmatically, while visually they can remain in the table to clarify what section content will be added to. Alternatively, they can be regular table rows with a single cell containing the Add button, which spans all table columns. 
* The content of the first cell of several of the tables (Column 1 header, "Main Body" in the chapters table) does not fit with the other content of the table and is confusing. If something is going to go in the table header it should be a column header. This content would ideally be moved out of the table entirely.
* The 'select/unselect all' table headers functionality is not well communicated to users. They function like buttons, but are communicated as links. We should more clearly indicate the user's ability to show/hide all elements in a column through a separate link/button (select all/deselect all which are always present to the user).
* Labels for checkboxes do not provide enough specificity about the checkbox. The label connects the checkbox with the column header but not the row header. 

**Expected behavior:** 
* Row headers should be clearly labeled as row headers. 
* Links that appear in the row headers should be moved to menu buttons in the row header cell. This way they will not be associated with every cell in the row.
* Data that does not belong to the tabular data of the table should be moved out of the table. This inlcudes the "Add X" buttons and the "Main Body" links in the chapters table. 

## TODO
- [ ] turn chapter names into row headers with roles & any other needed aria attributes
- [ ] insert permanent menu buttons for links edit | trash | view buttons (the ones which now only appear on hover)
- [ ] move 'add front matter' etc buttons out of table  
- [ ] Add permanent select all/deselect all buttons to the show in web / show in exports / show title columns
- [ ] replace part name with static column header called 'Chapter Title'
- [ ] move part name and actions outside of table + place 'add chapter' button next to them [no longer in table foot]
- [ ] Include information about the type of content (for example Dedication, Glossary, Demo Chapter, etc.) in the label. The following checkbox is communicated as “show in web”, instead it should be “Demo Chapter, Show in Web”. 
![Screenshot from 2021-10-12 15-34-19](https://user-images.githubusercontent.com/13485451/137037967-26c19d61-83e1-4ccf-af44-474b05c4edf8.png)

## Organize page
https://github.com/pressbooks/pressbooks/blob/dev/inc/admin/organize/namespace.php

**Comment by avaaec** [comment]
This is a UW ATS priority 1 issue

**Comment by SteelWagstaff** [comment]
TO DO: 
- [ ] turn chapter names into row headers with roles & any other needed aria attributes
- [ ] insert menu button for links edit | trash | view buttons which now appear on hover
- [ ] move 'add front matter' etc buttons out of table  
- [ ] replace part name with static column header called 'Chapter Title'
- [ ] move part name and actions outside of table + place 'add chapter' button next to them [no longer in table foot]

**Comment by ho-man-chan** [comment]
Mockup design: https://balsamiq.cloud/s1c84o8/pq304r6

Table Not Clicked
![image](https://user-images.githubusercontent.com/21694293/70088784-f2e45b00-15e4-11ea-9f19-6ab9eaccd59f.png)

Table Clicked
![image](https://user-images.githubusercontent.com/21694293/70088798-f7107880-15e4-11ea-89ce-ba40b0891091.png)

**Comment by avaaec** [comment]
HI @ho-man-chan! The UW ATX team will review this mockup and provide feedback asap.

**Comment by SteelWagstaff** [comment]
Hi @avaaec -- no urgency here. I just asked Ho Man to post mockup to document this as a work in progress. We'll give you more of a heads up before taking it on as an actual dev task in a sprint.

**Comment by SteelWagstaff** [comment]
Basalmiq link is no longer working. Will ask Ho Man where it exists?
